Calculator/Calculator/ViewController.swift

```diff
@@ -0,0 +1,333 @@
+//            let num4 = String(num3).components(separatedBy: ".")
+//
+//            if Int(num4[1] ) == 0 {
+//
+//                mainLabel.text = String(Int(num3))
+//            }else{
+//                mainLabel.text = String(num3)
+//
+//            }
+//        }else if ((mainLabel.text?.contains("+")) == true) {
+//            let number1 = mainLabel.text?.components(separatedBy: "+")
+//            let num1 = Double((number1?[0])!)
+//            let num2 = Double((number1?[1])!)
+//            let num3 = Double(num1!) + Double(num2!)
+//            let num4 = String(num3).components(separatedBy: ".")
+//
+//            if Int(num4[1] ) == 0 {
+//
+//                mainLabel.text = String(Int(num3))
+//            }else{
+//                mainLabel.text = String(num3)
+//            }
+//        }
+//        else{
+//            mainLabel.text?.append("/")
+//        }
+    }
+    
+    
+    @IBAction func multiplication(_ sender: Any) {
+        if mainLabel.text == "0"{
+            
+            
+        } else if mainLabel.text?.last == "/" || mainLabel.text?.last == "*" || mainLabel.text?.last == "+" || mainLabel.text?.last == "-" {
+            
+        }else{
+            mainLabel.text?.append("*")
+        }
+    }
+    
+    @IBAction func less(_ sender: Any) {
+        if mainLabel.text == "0"{
+            
+            
+        } else if mainLabel.text?.last == "/" || mainLabel.text?.last == "*" || mainLabel.text?.last == "+" || mainLabel.text?.last == "-" {
+            
+        }else{
+            mainLabel.text?.append("-")
+        }
+    }
+    
+    @IBAction func plus(_ sender: Any) {
+        if mainLabel.text == "0"{
+            
+            
+        } else if mainLabel.text?.last == "/" || mainLabel.text?.last == "*" || mainLabel.text?.last == "+" || mainLabel.text?.last == "-" {
+            
+        }else{
+            mainLabel.text?.append("+")
+        }
+    }
+    
+    @IBAction func ACBotton(_ sender: Any) {
+        mainLabel.text = "0"
+    }
+    
+    @IBAction func pointDecimal(_ sender: Any) {
+        if ((mainLabel.text?.contains(".")) == false){
+            mainLabel.text?.append(".")
+        }else{
+            mainLabel.text = mainLabel.text
+            
+        }
+    }
+    
+    @IBAction func nineNumber(_ sender: Any) {
+        if mainLabel.text == "0"{
+            mainLabel.text = "9"
+            
+        }else{
+            mainLabel.text?.append("9")
+        }
+    }
+    @IBAction func eightNumber(_ sender: Any) {
+        if mainLabel.text == "0"{
+            mainLabel.text = "8"
+            
+        }else{
+            mainLabel.text?.append("8")
+        }
+    }
+    @IBAction func sevenNumber(_ sender: Any) {
+        if mainLabel.text == "0"{
+            mainLabel.text = "7"
+            
+        }else{
+            mainLabel.text?.append("7")
+        }
+    }
+    
+    @IBAction func sixNumber(_ sender: Any) {
```

Comment:
you could use 1 function instead of 10 function, and use the info in the tag parameter of the button